### DISCUSSION TITLE
Introduce Owners Rotation

### DISCUSSION THREAD

**Discussion by qu1queee** [shipwright-io/community#60] [Ideas]
# Context

This is a follow-up on https://github.com/shipwright-io/community/issues/56#issuecomment-1031547934.

# Goal

[1] Speed-up the feedback-loop on contributions across Issues, Pull Requests. 
[2] Maintain the latest state of the project healthy(_e.g. triage nightly CI failures_)

# How to?

Introduce a rotation across [owners](https://github.com/shipwright-io/build/blob/main/OWNERS#L4-L11) with a _best-effort_ approach. The frequency of the rotation per individual tbd(daily, weekly). 

The rotation should consider the following(_at least at the beginning_):

- A runbook. To define a common approach for triage. 
- An slack channel to communicate ongoing rotation issues

# What's next?

- Define if this rotation applies to all projects under shipwright-io.

- Define the rotation participants and frequency.

- Define communication channels for the rotation.

- Define rotation hand-over. 

- Define the Runbook contents. For example:
   - How to triage Issues?
   - How to triage Pull Request?
   - How to debug CI issues?
   - Who to reach out if blocked?
 
- Which pieces could be automated?

# Additional Context

The above is based on [Tekton BuildCaptain Runbook Proposal](https://docs.google.com/document/d/1QJV0z2bMXdz_BZOkBwfxIP1BiktUb8c1lcifwqxF5wg/edit#heading=h.69dxtj1atzxu)

**Comment by qu1queee** [comment]
Follow up on the feedback from the BuildCaptain Runbook:

- [1] On making `Dependabot` alerts part of the rotation. There is a limitation on who can access the alerts. Only `maintainers` apparently. If we want this, we would need to document another way for everyone to see alerts.
- [2] On the need of introducing a `/support` on issues, to highlight the source of it. Also a `/grooming` level, to flag that the issue needs attention on the next grooming session.
- [3] For Stale issues, the Openshift CI should do the work, so less to do during rotation.
- [4] On the rotation schedule, a weekly start is proposed(1 person 1 full week).. best effort.
- [5] For rotation handovers, to do this during the Monday Community call.  Report in the way of a comment on the community meeting session.
- [6] On communications channel, we need a new channel to communicate things like [5] , Prs , etc. The current channel can remain as one intended for users to reach out to us.

**Comment by adambkaplan** [comment]
Closing this discussion as it is inactive.
